test/Writer/StreamTest.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DotTest\Log\Writer;
+
+use Dot\Log\Exception\InvalidArgumentException;
+use Dot\Log\Formatter\Json;
+use Dot\Log\Logger;
+use Dot\Log\Writer\Stream;
+use ErrorException;
+use PHPUnit\Framework\TestCase;
+
+use function curl_init;
+use function fopen;
+
+class StreamTest extends TestCase
+{
+    private Stream $subject;
+
+    private array $options = [
+        'stream'    => __DIR__ . '/../../log/error-log-{Y}-{m}-{d}.log',
+        'filters'   => [
+            'allMessages' => [
+                'name'    => 'priority',
+                'options' => [
+                    'operator' => '>=',
+                    'priority' => Logger::EMERG,
+                ],
+            ],
+        ],
+        'formatter' => [
+            'name' => Json::class,
+        ],
+    ];
+
+    /**
+     * @throws ErrorException
+     */
+    protected function setUp(): void
+    {
+        $this->subject = new Stream($this->options);
+    }
+
+    /**
+     * @throws ErrorException
+     */
+    public function testWillNotInstantiateWithInt(): void
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Resource is not a stream nor a string; received "integer"');
+        new Stream(47);
+    }
+
+    /**
+     * @throws ErrorException
+     */
+    public function testWillNotInstantiateWithWrongResource(): void
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Resource is not a stream nor a string; received "object"');
+        new Stream(['stream' => curl_init('url')]);
```

Comment:
```suggestion
        new Stream(['stream' => new \stdClass()]);
```
If all this test needs is an object, you can use an instance of any class you have at hand.
This way you can get rid of the curl-related additions.